src/core/Browser.js

```diff
@@ -13,54 +13,71 @@
+	// `true` for all browsers supporting [pointer events](https://msdn.microsoft.com/en-us/library/dn433244%28v=vs.85%29.aspx).
+	const pointer = !!window.PointerEvent;
 
-// @property touchNative: Boolean
-// `true` for all browsers supporting [touch events](https://developer.mozilla.org/docs/Web/API/Touch_events).
-// **This does not necessarily mean** that the browser is running in a computer with
-// a touchscreen, it only means that the browser is capable of understanding
-// touch events.
-const touchNative = 'ontouchstart' in window || !!window.TouchEvent;
+	// @property touchNative: Boolean
+	// `true` for all browsers supporting [touch events](https://developer.mozilla.org/docs/Web/API/Touch_events).
+	// **This does not necessarily mean** that the browser is running in a computer with
+	// a touchscreen, it only means that the browser is capable of understanding
+	// touch events.
+	const touchNative = 'ontouchstart' in window || !!window.TouchEvent;
 
-// @property touch: Boolean
-// `true` for all browsers supporting either [touch](#browser-touch) or [pointer](#browser-pointer) events.
-// Note: pointer events will be preferred (if available), and processed for all `touch*` listeners.
-const touch = touchNative || pointer;
+	// @property touch: Boolean
+	// `true` for all browsers supporting either [touch](#browser-touch) or [pointer](#browser-pointer) events.
+	// Note: pointer events will be preferred (if available), and processed for all `touch*` listeners.
+	const touch = touchNative || pointer;
 
-// @property retina: Boolean
-// `true` for browsers on a high-resolution "retina" screen or on any screen when browser's display zoom is more than 100%.
-const retina = (window.devicePixelRatio || (window.screen.deviceXDPI / window.screen.logicalXDPI)) > 1;
+	// @property retina: Boolean
+	// `true` for browsers on a high-resolution "retina" screen or on any screen when browser's display zoom is more than 100%.
+	const retina = (window.devicePixelRatio || (window.screen.deviceXDPI / window.screen.logicalXDPI)) > 1;
 
-// @property mac: Boolean; `true` when the browser is running in a Mac platform
-const mac = navigator.platform.startsWith('Mac');
+	// @property mac: Boolean; `true` when the browser is running in a Mac platform
+	const mac = navigator.platform.startsWith('Mac');
 
-// @property mac: Boolean; `true` when the browser is running in a Linux platform
-const linux = navigator.platform.startsWith('Linux');
+	// @property mac: Boolean; `true` when the browser is running in a Linux platform
+	const linux = navigator.platform.startsWith('Linux');
+
+	return 	{
+		chrome,
+		safari,
+		mobile,
+		pointer,
+		touch,
+		touchNative,
+		retina,
+		mac,
+		linux
+	};
+}
 
 function userAgentContains(str) {
 	return navigator.userAgent.toLowerCase().includes(str);
 }
 
 
-export default {
-	chrome,
-	safari,
-	mobile,
-	pointer,
-	touch,
-	touchNative,
-	retina,
-	mac,
-	linux
-};
+export default setupBrowser();

```

Comment:
This breaks tree-shaking (which, from my PoV, is more important than SSR friendlyness)

Reply:
Good callout. I'll see if there's a way to have both